### ISSUE TITLE
Prompt Tuning documentation trains all parameters

### ISSUE THREAD

**Issue by Ben-Epstein** [huggingface/peft#1284]
### System Info

Google colab

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder
- [ ] My own task or dataset (give details below)

### Reproduction

It seems like the [documentation for prompt tuning](https://huggingface.co/docs/peft/task_guides/clm-prompt-tuning) sets up the optimizer incorrectly
`optimizer = torch.optim.AdamW(model.parameters(), lr=lr)` 

Shouldn't this be only the trainable parameters? Otherwise, you're using a lot of additional memory for params that won't be tuned. In this example, you tune on bloom with 500M parameters. 

AdamW requires 18 bytes per model parameter, so 18B * 500M = 9GB. Is this correct, or is there a reason the optimizer requires all parameters? 

Thanks!

### Expected behavior

Potentially a new function `model.trainable_parameters` that would then be passed in

**Comment by younesbelkada** [comment]
Hi @Ben-Epstein 
Thanks for noticing this bug!
I think it can be modified by:
```diff
+ optimizer = torch.optim.AdamW(filter(lambda p: p.requires_grad, model.parameters()), lr=lr)
- optimizer = torch.optim.AdamW(model.parameters(), lr=lr)
```
Would you be happy to open a PR for fixing this?

**Comment by Ben-Epstein** [comment]
> Hi @Ben-Epstein 
> Thanks for noticing this bug!
> I think it can be modified by:
> ```diff
> + optimizer = torch.optim.AdamW(filter(lambda p: p.requires_grad, model.parameters()), lr=lr)
> - optimizer = torch.optim.AdamW(model.parameters(), lr=lr)
> ```
> Would you be happy to open a PR for fixing this?

Sure thing, https://github.com/huggingface/peft/pull/1285

**Comment by younesbelkada** [comment]
Thanks a lot!

**Comment by BenjaminBossan** [comment]
Hey @Ben-Epstein, thanks for reporting this.

> Shouldn't this be only the trainable parameters? Otherwise, you're using a lot of additional memory for params that won't be tuned. In this example, you tune on bloom with 500M parameters.

Did you check whether filtering of parameters makes any difference in practice? I think PyTorch is already quite efficient of taking care of that under the hood.

**Comment by yunfeng-eiq** [comment]
Interesting. We tried it, and indeed, it didn't seem to make much difference in terms of memory consumption. So you're probably right that pytorch already optimized it under the hood.

**Comment by EeyoreLee** [comment]
Pytorch optim has a hook named `_init_group` which will auto filter the param without gradient. Both of these two approach will do the same action.
